### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by uzairali19**
This branch is to update and delete tasks dynamically. This enables the user to input a task and then displays that instead of hard-coded data.

### CODE REVIEW

File: src/status.js [hunk 1]
```diff
@@ -1,65 +1,97 @@
...
-          list.appendChild(li);
-        }
+class Status {
+  constructor(form, todoTask, mainList, list, data, completed) {
+    this.form = form;
+    this.todoTask = todoTask;
+    this.mainList = mainList;
+    this.list = list;
+    this.data = data;
+    this.completed = completed;
```
Comment (afanimartin) [inline src/status.js:8]: - A todo item should not be false by default. Please assign a default false value to the propery. `this.completed=false`

File: src/status.js [hunk 2]
```diff
@@ -1,65 +1,97 @@
...
+        newTask.splice(i, 1);
+        localStorage.setItem('task', JSON.stringify(newTask));
+      });
+    });
   }
 
-  updateCheckbox(todoList) {
-    const checkBox = document.querySelectorAll('.check-input');
-    checkBox.forEach((v, i) => {
+  completeTodo() {
```
Comment (afanimartin) [inline src/status.js:79]: - Add a function for editing a todo item. Make it easier to correct mistakes made in creating a todo item.

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @uzairali19, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/status.js (hunk 1 above)
Comment (uzairali19) [reply src/status.js:8]: Hello, Thank you for such a prompt review.

I'll urge you to double-check this, please, the text that you have highlighted should've been used as an argument but I am not using that. So, I am going to remove that now. 
Plus it clearly says in the requirements that the property should be set to `false` and the `index` should be the length of node items. I have just manipulated data to make it more dynamic.

Screenshot 2021-10-30 at 9 48 10 PM

File: src/status.js (hunk 2 above)
Comment (uzairali19) [reply src/status.js:79]: Can you please check, I already have a function for adding a to-do item? My `completeTodo()` function is only meant for the `<checkbox>.checked` value. Which is returning either true or false.

File: src/status.js [hunk 3]
```diff
@@ -1,65 +1,96 @@
...
+  removeData() {
+    const del = document.querySelectorAll('#del');
+    del.forEach((v, i) => {
+      v.addEventListener('click', (e) => {
+        e.preventDefault();
+        const div = v.parentElement;
+        const li = div.parentElement;
+        li.remove();
+        const newTask = JSON.parse(localStorage.getItem('task'));
+        newTask.splice(i, 1);
```
Comment (EricMbouwe) [inline src/status.js:72]: - [ ] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. Currently, the function is not working as expected. After a click to remove the first element of the list, everything is wiped. Kindly double-check this and fix that issue

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @uzairali19,

Great work on making the changes requested by a previous reviewer 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

# Required Changes ♻️ ⛔️
- [ ] Currently, there is no way to `edit` a task, and we can't find a function for it in the code as well. You need to implement a function for editing task descriptions as this is part of the requirements.
- [ ] A function for the "Clear all completed" button using the `Array.filter()` method should be implemented as well. Actually, we can't delete all completed todos. Kindly add that feature

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**